### PR TITLE
Roll all logs by default

### PR DESCRIPTION
**Pull request by julianvmodesto**
### Issue
#1372 

### TODO
- [x] rip out nested block parsing
- [x] add new subdirectives + share between logging middleware
- [x] set defaults to rotate logs
- [ ] update docs

work in progress!

### CODE REVIEW

**Comment by julianvmodesto** [comment]
Ok, looks like this implements the desired behavior for default rolling logs config.

```
# old
log ... {
    rotate {
        size maxsize
        age maxage
        keep maxkeep
    }
}

# new
log ... {
    rotate_size maxsize
    rotate_age maxage
    rotate_keep maxkeep
}

# old
errors {
    log error.log {
      rotate_size maxsize
      rotate_age maxage
      rotate_keep maxkeep
    }
}

# new
errors {
    log error.log {
      rotate_size maxsize
      rotate_age maxage
      rotate_keep maxkeep
    }
}
```

Does this PR do enough to rip out the original nested block parsing for `log`?

The `log` directive was flattened by removing the `rotate` subdirective, but `c.IncrNest()` seems like it's still the best option for parsing both the `log` and `errors` directives.

See here: https://github.com/mholt/caddy/pull/1379/files#diff-b44dbfe18c42f72684e2df145fae3da9L92

**Comment by mholt** [comment]
Thanks for starting on this!

Let's flatten this:
```
errors {
    log ... {
      rotate_size maxsize
      rotate_age maxage
      rotate_keep maxkeep
    }
}
```

down to this:

```
errors {
      rotate_size maxsize
      rotate_age maxage
      rotate_keep maxkeep
}
```

We can get rid of the whole IncrNest() function I think, since nesting isn't really a thing without this; it won't be needed.

**Comment by julianvmodesto** [comment]
Got it, flattened + removed IncrNest()

**Comment by mholt** [comment]
Nice work :+1: Thanks. I'm away for a week but will get back to this ASAP!

File: caddyhttp/httpserver/roller.go [hunk 1]
```diff
@@ -4,10 +4,6 @@ import (
 	"io"
 	"path/filepath"
 	"strconv"
-
-	"github.com/mholt/caddy"
-
-	"gopkg.in/natefinch/lumberjack.v2"
```
Comment (mholt) [inline caddyhttp/httpserver/roller.go]: Oops, this import needs to be restored

Comment (julianvmodesto) [inline caddyhttp/httpserver/roller.go]: Agh, re-imported!

**Comment by julianvmodesto** [comment]
Hm, anything I should do to fix the AppVeyor build?

**Comment by mholt** [comment]
That's weird. GitHub bug? It definitely passed: https://ci.appveyor.com/project/mholt/caddy/build/2195

I'll merge it in soon. :) I'm at school for a bit.

**Comment by mholt** [comment]
@julianvmodesto Can you resolve the merge conflicts?

And before I merge, have you verified that there is/are test cases to ensure logs are rolled by default and the default configuration is properly overwritten when specified?

**Comment by julianvmodesto** [comment]
Conflicts resolved!

Yes – examples of test cases for the default + overrides are here:

- Error log default
  - directive: `errors errors.txt`
  - [test case](https://github.com/mholt/caddy/pull/1379/files#diff-1f8b3e26f9d0408cac17181fca7fce49L63)
- Error log with rotation specified
  - directive: `errors errors.txt { rotate_size 2 rotate_age 10 rotate_keep 3 }`
  - [test case](https://github.com/mholt/caddy/pull/1379/files#diff-1f8b3e26f9d0408cac17181fca7fce49R88)
- Log default
  - directive: `log`
  - [test case](https://github.com/mholt/caddy/pull/1379/files#diff-158e3f59badf8ac3fadd84e23ae55bbcL55)
- Log with rotation specified
  - directive: `log access.log { rotate_size 2 rotate_age 10 rotate_keep 3 }`
  - [test case](https://github.com/mholt/caddy/pull/1379/files#diff-158e3f59badf8ac3fadd84e23ae55bbcR177)

**Comment by mholt** [comment]
Great work! Thanks @julianvmodesto. I'll merge this now and invite you to be a collaborator to the project so you can help by reviewing PRs, merging accepted ones, and contributing fixes on branches instead of forks. Really appreciate your contribution!

**Comment by wendigo** [comment]
IncrNest was removed from Controller which breaks caddy filter building :(

**Pull request merged**